README.md

```diff
@@ -0,0 +1,152 @@
+# Plotly
+
+This library wraps the [Plotly](https://plot.ly/) [REST API] (https://plot.ly/rest/), allowing you to graph and style data obtained from Imp-connected sensors.
+
+This class allows for simple creation of time-series data graphs while exposing access for styling graphs using all features of the Plotly API.  Note that this library requires creation of a Plotly account.
+
+### Callbacks
+Almost all methods in this class (including the constructor) take an optional *callback* argument.
+This is a function that takes arguments *response* and *plot*, where *response* is a table representing a response from the Plotly servers and *plot* is a reference to the plot object.  The *response* object mirrors that provided in the callback to [httprequest.sendasync()](https://electricimp.com/docs/api/httprequest/sendasync/) with the addition of a *decoded* field that contains the JSON body of the response in a Squirrel table.
+
+**Note that while the constructor will return immediately, it is only safe to operate on the resulting object once the callback has been called.**  The *plot* argument to the callback is provided for this purpose.  It is also the user's responsibility to ensure at this step that the construction has succeeded by checking the HTTP response code and/or Plotly response messages.
+
+## Constructor: Plotly(*userName, userKey, FileName, worldReadable, traces [, callback]*)
+
+To create a plot, you need to call the constructor with your Plotly authentication information and some basic data about the new graph.
+
+To find your *userName* and *userKey*, go to the Plotly settings and copy the Username and API key as highlighted below.  Note that the *userKey* is **not** your password, but is an API key that Plotly provides for developers.  Whenever you cycle your API key (e.g. by clicking "Generate a new key"), you will have to update this value in your code as well.
```

Comment:
The first mention of "Plotly settings" should link to the appropriate page if possible